src/Tasks/Microsoft.Common.props

```diff
@@ -148,6 +148,16 @@ Copyright (C) Microsoft Corporation. All rights reserved.
 
   <Import Project="$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.VisualStudioVersion.v*.Common.props" Condition="'$(VisualStudioVersion)' == ''" />
 
+  <!--
+    Properties for extension of ProjectReferenceTargets.
+    Append any current value which may have been provided in a Directory.Build.props since the intent was likely to append, not prepend.
```

Comment:
The exact location of this seems important.

If appended to in Directly.Build.props, this logic will append what the user did instead of append our stuff. The assumption being that usually, people would want to extend stuff after build, not before (???)

Also, if appending to `ProjectReferenceTargetsForBuild` (or the Clean one), this *will* affect Rebuild if done early enough, but not if done later. If done later, they'd need to append to the Rebuild one as well. This I guess would only be for folks who did it directly in their project, before the common targets.

The project extensions (like nuget.g.props) happens before this so has the same behavior as doing it in Directory.Build.props.

If extending these in directory.build.props, I *think* it's too late.

Reply:
This seems right to me. @rainersigwald ?

Reply:
Filed https://github.com/Microsoft/msbuild/issues/3808 to keep track of things like this that will eventually need to be documented.